lib/matplotlib/widgets.py

```diff
@@ -485,21 +514,290 @@ def on_changed(self, func):
+            self.set_max(val)
+
+    def _update(self, event):
+        """Update the slider position."""
+        if self.ignore(event) or event.button != 1:
+            return
+
+        if event.name == "button_press_event" and event.inaxes == self.ax:
+            self.drag_active = True
+            event.canvas.grab_mouse(self.ax)
+
+        if not self.drag_active:
+            return
+
+        elif (event.name == "button_release_event") or (
+            event.name == "button_press_event" and event.inaxes != self.ax
+        ):
+            self.drag_active = False
+            event.canvas.release_mouse(self.ax)
+            return
+        if self.orientation == "vertical":
+            val = self._update_val_from_pos(event.ydata)
+        else:
+            val = self._update_val_from_pos(event.xdata)
+        if val not in [None, self.val]:
+            self.set_val(val)
+
+    def _format(self, val):
+        """Pretty-print *val*."""
+        if self.valfmt is not None:
+            return (self.valfmt % val[0], self.valfmt % val[1])
+        else:
+            s1 = super()._format(val[0])
+            s2 = super()._format(val[1])
+            # fmt.get_offset is actually the multiplicative factor, if any.
+            return (s1 + self._fmt.get_offset(), s2 + self._fmt.get_offset())
+
+    def set_min(self, min):
+        """
+        Set the lower value of the slider to *min*
 
         Parameters
         ----------
-        cid : int
-            Connection id of the observer to be removed
+        min : float
         """
-        self._observers.disconnect(cid)
+        self.set_val((min, self.val[1]))
 
-    def reset(self):
-        """Reset the slider to the initial value"""
-        if self.val != self.valinit:
-            self.set_val(self.valinit)
+    def set_max(self, max):
+        """
+        Set the lower value of the slider to *max*
+
+        Parameters
+        ----------
+        max : float
+        """
+        self.set_val((self.val[0], max))
+
+    def set_val(self, val):
+        """
+        Set slider value to *val*
+
+        Parameters
+        ----------
+        val : tuple or arraylike of float
+        """
+        val = np.sort(np.asanyarray(val))
+        if val.shape != (2,):
+            raise ValueError(
+                "val must have shape (2,)" f"but has shape {val.shape}"
+            )
+        val[0] = self._min_in_bounds(val[0])
+        val[1] = self._max_in_bounds(val[1])
+        xy = self.poly.xy
+        if self.orientation == "vertical":
+            xy[4] = 0, val[0]
+            xy[0] = 0, val[0]
+            xy[1] = 0, val[1]
+            xy[2] = 1, val[1]
+            xy[3] = 1, val[0]
+        else:
+            xy[4] = val[0], 0
+            xy[0] = val[0], 0
+            xy[1] = val[0], 1
+            xy[2] = val[1], 1
+            xy[3] = val[1], 0
+        self.poly.xy = xy
+        fmted = self._format(val)
+        self.valtext.set_text(rf"({fmted[0]}, {fmted[1]})")
```

Comment:
Again, already a tuple.
```suggestion
        self.valtext.set_text(str(self._format(val)))
```